packages/contentstack-clone/src/commands/cm/stacks/clone.js

```diff
@@ -60,8 +68,24 @@ class StackCloneCommand extends Command {
         await cloneHandler.start();
         let successMessage = 'Stack cloning process have been completed successfully';
         await this.cleanUp(pathdir, successMessage);
+      }
+
+      if (sourceManagementTokenAlias && destinationManagementTokenAlias) {
+        if (sourceStackBranch || targetStackBranch) {
+          if (_authToken) {
+            handleClone()
+          } else {
+            console.log("AuthToken is not present in local drive, Hence use 'csdx auth:login' command for login")
+            this.exit(1)
+          }
+        } else {
+          handleClone()
+        }
+      } else if (_authToken) {
+        handleClone()
       } else {
         console.log("AuthToken is not present in local drive, Hence use 'csdx auth:login' command for login");
```

Comment:
"AuthToken is not present in local drive, Hence use 'csdx auth:login' command for login" - Can you use better message here. Please check other places where we enforce log in. Please ask user to login to execute this command and provide the login command. "Please login to execute this command, csdx auth:login" - check other places